src/components/SlidePanel/SlidePanel.js

```diff
@@ -1,23 +1,10 @@
-import { cloneElement } from "react";
 import classnames from "classnames";
 import Spinner from "@canonical/react-components/dist/components/Spinner";
 
 import useEventListener from "hooks/useEventListener";
 
 import "./_slide-panel.scss";
 
-function generatePanelContent(isLoading, children, onClose) {
```

Comment:
The usage of `SlidePanel` is going to change from being a wrapper of child content to being included as part of that child content itself. This allows much easier control over what elements respond to the 'click outside' and the ability to interact with the various closure mechanisms of the panel.